### DISCUSSION TITLE
Use PHP attributes for enum labels

### DISCUSSION THREAD

**Discussion by andy3471** [filamentphp/filament#14204] [Ideas]
Currently when using enums with Filament, there is the HasLabel contract, which works like this:

```
<?php

namespace Bullet\Translator\Enums;

use Filament\Support\Contracts\HasLabel;

enum LanguageDetectionProvider: string implements HasLabel
{
    case AiCompletion = 'ai-completion';

    case Local = 'local';

    case Google = 'google';

    case Default = 'default';

    public function getLabel(): string
    {
        return match ($this) {
            self::AiCompletion => 'AI Completion',
            self::Local        => 'Local',
            self::Google       => 'Google',
            self::Default      => 'Default',
        };
    }
}
```

I think it would be simpler and cleaner if we could use PHP attributes to define labels:

```
<?php

namespace Bullet\Translator\Enums;

use Filament\Support\Contracts\HasLabel;
use Filament\Attributes\Enums\Label;

enum LanguageDetectionProvider: string implements HasLabel
{
    #[Label('AI Completion')]
    case AiCompletion = 'ai-completion';

    #[Label('Local')]
    case Local = 'local';

    #[Label('Google')]
    case Google = 'google';

    #[Label('Default')]
    case Default = 'default';
}
```

We can always default to the original syntax if they require more granular control.

I can implement this and PR it into V4 if this is agreed.

**Comment by dissto** [comment]
I personally don't think this is pretty and has no obvious benefit over the traditional way.

Though you would have to do the same with `getColor` `getIcon` and `getDescription`...and what about translating for example the label and the description? Can you pass a call to `__()` into an attribute?🤔

**Reply by andy3471** [reply]
I could implement colour, icon and description as well. 

We can't directly pass a call to `__()` in an attribute, but we could do something like:
`#[Label(label: 'Ai Completion', translated: true)]` or `#[TranslatedLabel('Ai Completion')]`
